crates/iota-core/src/authority/shared_object_congestion_tracker.rs

```diff
@@ -971,10 +1004,10 @@ mod object_cost_tests {
 
         // Set max_execution_duration_per_commit to only allow 1 transaction
         // to go through.
-        let max_execution_duration_per_commit = match mode {
+        let (max_execution_duration_per_commit, max_overshoot_per_commit) = match mode {
             PerObjectCongestionControlMode::None => unreachable!(),
-            PerObjectCongestionControlMode::TotalGasBudget => 12,
-            PerObjectCongestionControlMode::TotalTxCount => 3,
+            PerObjectCongestionControlMode::TotalGasBudget => (12, 0),
+            PerObjectCongestionControlMode::TotalTxCount => (3, 0),
```

Comment:
```suggestion
            PerObjectCongestionControlMode::TotalTxCount => (1, 0),
```
Why do we have this set to 3 instead of 1? The comment says we want to only allow 1 tx to go through. Is the comment misleading? The test passes with 1, btw.

Reply:
The reason the limit is 3 is because we also set some initial objects in the tracker when we initialise it below. In the `TotalTxCount` mode, we have a transaction with end time 2 when we initialise it. Then when we bump execution slots below that, we add the next transaction with `start_time` 2, so it's end time is 3. This is the one transaction that can fit. 

The reason it works when you set this value to 1 is that the initialiser and bump function don't care about the congestion limit, so we can still fill the tracker beyond the limit. THe only thing checked in the test is that something gets deferred and returns shared objects.

Reply:
Thanks, I took me a while to understand this test, even though it is not directly part of the PR.

Why I found the test flow confusing:

1. First, we have this

   ```rust
        let tx_gas_budget = 5;

        // Set max_execution_duration_per_commit to only allow 1 transaction
        // to go through.
        let (max_execution_duration_per_commit, max_overshoot_per_commit) = match mode {
            PerObjectCongestionControlMode::None => unreachable!(),
            PerObjectCongestionControlMode::TotalGasBudget => (12, 0),
            PerObjectCongestionControlMode::TotalTxCount => (3, 0),
        };
   ```

2. Then, we initialize the tracker with initial values for `object 0` being `9` in gas budget mode and `2` in tx count mode.

3. From this context, it is easy to see that one transaction is allowed to go through (as the comment says) in the tx count mode. But in the gas budget mode, how can a transaction with `tx_gas_budget = 5` go through if the limit is `12` and initial end time is set to `9`? It turns out that `tx_gas_budget = 5` is only used for a transaction which we don't want to go through (w.r.t. `object 0`), while the gas budget of `1` is used for a transaction we want to go through. But this part is not clear until:

   ```rust
        // add a transaction that writes to object 0 and 1.
        let tx = build_transaction(
            &[(shared_obj_0, true), (shared_obj_1, true)],
            1,
            TEST_ONLY_GAS_PRICE,
        );
   ```

Moving the `let tx_gas_budget = 5;` statement closer to where the value is used, adding a comment to it, adding names for "magic" numbers like `9`, `2`, `1` could improve the readability a lot.

Reply:
That is indeed a bit confusing, I will do as you suggested, naming the numbers where possible and initalising in the place where they're used.